### DISCUSSION TITLE
Timeout reached, aborting! It fails when I try Build and deployment.

### DISCUSSION THREAD

**Discussion by MutianGao** [community/community#75964] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Question

### Body

Hey guys, I'm a new github user and in the recent past I've started to learn something about html. Now I would like to Build and deploy a website to submit as a homework  (https://github.com/MutianGao/MutianGao.github.io)

I've looked for some tutorials to learn how to do this, but this keeps showing a timeout. Does anyone know how to fix this? I think I'm getting it wrong somewhere.

**Comment by ghost** [comment, marked as answer]
Hello @MutianGao 

I have checked actions in your repo and it is default limitation that deployment will be rejected if it takes more than 10 min. your all deployments took around 10min and 20-40 sec, so github stopped deployment. 

you should check why it is taking so long time to deploy it. its worth to mention that github also allows 10 deployment per hour only (but not sure if this is per user or per repo). I have checked all your 3 commits and it seems that you have changed nothing (literally nothing) between first and last commit, so since it deploied first time it should not be your (your code) fault. i think its one of the githubs limitations for using github pages. 

I think its also good idea to remove all the giffs from the repo and try do deploy then (create copies first).

Also before you do anything else i think you should also try to use custom workflow to deploy repo since it allows more controll and github has less limitations on that. Also since you said you are learning some staff on github its also worth to learn Custom Workflows. you can find more [there](https://docs.github.com/en/pages/getting-started-with-github-pages/using-custom-workflows-with-github-pages)

Also you should probably try to redeploy your repo after 1-2 hour (if its github limitations it will probably work)

Hope this helps

**Reply by MutianGao** [reply]
I really appreciate your willingness to help.
On the earliest few submissions, I forgot to change the paths to some files and didn't set the home page to index.html. so in fact those appeared to submit nothing(?), and the page was in a 404 state at the time.
When I tried again this morning, it only took about 45S to submit successfully, which is very interesting.That Custom Workflows tutorial also seems to be very useful, but I haven't quite gotten through it yet. XD
But anyways, thanks again for your help.

**Reply by ghost** [reply]
no problem.

Glad you have fixed your problem. since it worked today with no problem i guess it was one of github pages limitations as i said before.

one thing you can do is to avoid deploiment all the time when you change small thing in your repo. probably this can be achived in many different ways but simplest one is to create different branch for work. Right now for your repo deployment starts when changes are made on "master" branch (its default). you can create for example "dev" branch (name does not matter), work on this "dev" branch (made commits and needed changes) and once you are done you can create pull request from "dev" to "master" and merge it. merge event will start depoyment on its own. this way you will be able to monitor your code better and avoid deploying broken code.
